pkg/descheduler/strategies/nodeutilization/highnodeutilization.go

```diff
@@ -29,8 +29,48 @@ import (
 	"sigs.k8s.io/descheduler/pkg/descheduler/evictions"
 	nodeutil "sigs.k8s.io/descheduler/pkg/descheduler/node"
 	podutil "sigs.k8s.io/descheduler/pkg/descheduler/pod"
+	"sigs.k8s.io/descheduler/pkg/utils"
 )
 
+// balanceNodes balance nodes from low utilized nodes to high utilized nodes. A corner case: All nodes are underutilized
+func balanceNodes(sourceNodes, destinationNodes []NodeInfo) ([]NodeInfo, []NodeInfo) {
+	if len(sourceNodes) == 0 || len(destinationNodes) > 0 {
+		return sourceNodes, destinationNodes
+	}
+	numOfBalanced := 0
+	partition := int(float64(len(sourceNodes)) * 0.5)
+	src := make([]NodeInfo, 0, partition)
+	dst := destinationNodes
+	hasImportantPods := func(node *NodeInfo) bool {
+		for _, pod := range node.allPods {
+			if utils.IsMirrorPod(pod) {
+				return true
+			}
+			if utils.IsPodWithPVC(pod) {
+				return true
+			}
+			if utils.IsCriticalPriorityPod(pod) {
+				return true
+			}
+		}
+		return false
+	}
+	for _, node := range sourceNodes {
+		if nodeutil.IsNodeUnschedulable(node.node) {
+			src = append(src, node)
+			continue
+		}
+
+		if numOfBalanced < partition && !hasImportantPods(&node) {
```

Comment:
With `!hasImportantPods(&node)` as a condition is the goal here to protect nodes that have important pods from attracting pods from the source nodes? Given the kube-scheduler does not know anything about hasImportantPods property, this might not actually work as expected. Also, when all the nodes have at least one important node, dst will be empty.

Reply:
I'm not sure how to split the low utilization nodes into two is the best way